lib/tsp_helper.rb

```diff
@@ -36,7 +36,7 @@ def self.create_tsp(vrp, vehicle)
     }
 
     problem = {
-      matrices: vrp.matrices,
+      matrices: [Models::Matrix.find_by_id(vehicle.matrix_id)],
```

Comment:
by using, `Models::X.find_methods` we could find/retrieve objects in memory which are not from "the VRP" used by the current request/worker. To avoid this problem, I suggest to always retrieve sub-models from parent model

Reply:
As far as I know, the workers do not share the active_hash objects (and its database) and each worker has a unique active_hash object (and its database). I suspect this might not be the case for server threads. We now make sure that there is only the current VRPs loaded in the worker memory by loading the object into the active_hash database as well. 

The risque I see is that if a function that uses a `find_by_id` logic is called via server, it might be problematic (for example, if someday a refacto moves some parse_output logic to server side but I don't know if we should worry about that).

Reply:
Think about our API can be called without worker (synchronously: true), in this case memory is shared by every request

Reply:
Currently each server and worker thread have its own active_hash database. For server threads, it seems to be the default behavior of nginx/passenger, while locally it requires a config file for puma.

Reply:
Recap of Bdx discussion: this is not an issue at the moment because due to active_hash, we are forced to keep the nginx workers separate -- i.e., each nginx worker treats a single request so they don't share memory. We said we shall talk during daily point what to do. 

I suggest keeping @braktar modifications since it is less typing and if we ever could treat multiple requests per worker, this means we have fixed active_hash issue, which means find_by_id's are also fixed (since they come from active_hash).